.github/workflows/ci.yml

```diff
@@ -0,0 +1,63 @@
+# Name of the workflow
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master and stable branches
+on:
+  push:
+    branches: [ master, stable ]
+  pull_request:
+    branches: [ master, stable ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Main building and testing
+  build-test:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Define the platforms to run on. Later determines the Docker image that'll be used
+    strategy:
+      matrix:
+        platform: [ c7-dev-py2, c7-dev, c8-dev, u18-dev, u20-dev ]
+
+    # Sets up environment variables
+    env:
+      PLATFORM: ${{ matrix.platform }}
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+    # Checks-out our repository under $GITHUB_WORKSPACE, so our job can access it
+    - uses: actions/checkout@v2
+
+    # Pulls the associated Docker image
+    - name: Docker pull
+      run: |
+        if [ "$PLATFORM" == "c7-dev-py2" ];
+        then
+          docker pull hepsoftwarefoundation/c7-dev
```

Comment:
Could it be made a little better here if we defined `DIMAGE` as the docker image to pull? And for `c7-dev-py2` also defined at this point `CMAKE_EXTRA=-DPYTHON_TEST=python2`? It seems a bit of a cleaner separation between PLATFORM and Docker image.

```
if [ "$PLATFORM" == "c7-dev_py2" ];
then
  DIMAGE=hepsoftwarefoundation/c7-dev
  CMAKE_EXTRA=-DPYTHON_TEST=python2
else
  DIMAGE=hepsoftwarefoundation/$PLATFORM
fi
docker pull $DIMAGE
```

Reply:
This logic is updated now. Setting environment variables in different steps is a bit weird if you want to carry it over to other steps. But, in essence we do exactly this now.

Reply:
One thing that grates for me still is the way we set specific environment variable in the `docker run` command, but I can't quite see an easy way out of that right now 🤔 

I think this, at least, means that in one place it's clear what `_py2` means, which is a good thing.